packages/core/src/lib/map/map.ts

```diff
@@ -164,19 +165,37 @@ export function mapMutate<
   errorHandler: ErrorHandler,
   destinationObj: TDestination
 ): void {
-  map(
+  map({
     sourceObj,
     mapping,
     options,
     mapper,
     errorHandler,
-    (destinationMember: string[]) => (value: unknown) => {
+    setMemberFn: (destinationMember: string[]) => (value: unknown) => {
       if (value === undefined) {
         return;
       }
       setMutate(destinationObj, destinationMember, value);
+    },
+    getMemberFn: (memberPath: string[] | undefined) => {
+      return get(destinationObj, memberPath) as Record<string, unknown>;
     }
```

Comment:
suggestions: looks like we're reusing this for `mapMutate` case. Let's make two functions for `setMemberFn` and `getMemberFn` to reuse them in the `map()` body?

Reply:
Sorry, i didn't get what you mean.

Reply:
Oh, you mean extract L:175 ~ L:178 and L:181 to function and use it in `mapMutate` and `map` L:361 right?

Reply:
It's something like this

```ts
function setMemberMutateFn(destinationValue: unknown) {
   return (destinationMemberPath: string[]) => (value: unknown) => {
      if (value !== undefined) {
         setMutate(destinationValue, destinationMemberPath, value);
      }
   }
}

function getMemberMutateFn(destinationValue: unknown) {
   return (memberPath: string[] | undefined) => {
      return get(destinationValue, memberPath) as Record<string, unknown>
   }
}
```

then in `mapMutate()` 

```ts
map({
   /** ... */
   setMemberFn: setMemberMutateFn(destinationObj),
   getMemberFn: getmemberMutateFn(destinationObj)
})
```

in `map()` where you have `// nested mutate`, you then can:

```ts
if (destinationMemberValue !== undefined) {
   map({
      /** ... */
      setMemberFn: setMemberMutateFn(destinationMemberValue),
      getMemberFn: getMemberMutateFn(destinationMemberValue)
   );
   continue;
}
```

Reply:
Done in new commit.